### PR TITLE
Align naming of config options

### PR DESCRIPTION
**Pull request by dbenage-cx**
Superceeds #1533 

[Forum thread](http://freeorion.org/forum/viewtopic.php?f=6&t=10752)

[Comparison of most options](https://gist.github.com/dbenage-cx/37ff496e99a10b9c0e2e1518a7a90e46)

### CODE REVIEW

**Comment by geoffthemedio** [comment]
rebase?

**Comment by dbenage-cx** [comment]
Rebased

**Comment by geoffthemedio** [comment]
again?

**Comment by dbenage-cx** [comment]
Rebased

**Comment by geoffthemedio** [comment]
@Dilvish-fo any concerns?

**Comment by Dilvish-fo** [comment]
The AI related changes look reasonable to me.

File: AI/AIInterface.h [hunk 1]
```diff
@@ -264,7 +264,7 @@ namespace AIInterface {
 
     /** @brief Return the canonical AI directory path
      *
-     * The value depends on the ::OptionsDB `resource-dir` and `ai-path` keys.
+     * The value depends on the ::OptionsDB `content.path` and `ai-path` keys.
```
Comment (geoffthemedio) [inline AI/AIInterface.h]: not a fan of this change... "content" and "resource" are not the same thing.

File: UI/BuildDesignatorWnd.cpp [hunk 2]
```diff
@@ -35,8 +35,9 @@ namespace {
     const std::string PROD_SELECTOR_WND_NAME = "production.selector";
     const std::string PROD_SIDEPANEL_WND_NAME = "production.sidepanel";
 
-    void    AddOptions(OptionsDB& db)
-    { db.Add("UI.windows." + PROD_PEDIA_WND_NAME + ".persistently-hidden", UserStringNop("OPTIONS_DB_PRODUCTION_PEDIA_HIDDEN"), false, Validator<bool>()); }
+    void    AddOptions(OptionsDB& db) {
+        db.Add("ui." + PROD_PEDIA_WND_NAME + ".hidden.enabled", UserStringNop("OPTIONS_DB_PRODUCTION_PEDIA_HIDDEN"), false, Validator<bool>());
```
Comment (geoffthemedio) [inline UI/BuildDesignatorWnd.cpp]: prefer `{}` on same line for single line functions

File: AI/AIInterface.h (hunk 1 above)
Comment (Dilvish-fo) [reply AI/AIInterface.h]: Well, the AI aspect of it, where you've noted your comment here, is just harmonizing the AI interface comments with the changes this PR makes elsewhere, which on its own seems appropriate to me.

If your main concern is about the underlying change of the option name from 'resource-dir' to 'content.path', well, @dbenage-cx might have his own arguments to make, but I'll go ahead and opine that the change seemed reasonable to me-- the word 'resource' is indeed a bit broader than the word 'content', but the more specific word 'content' seems reasonable to me in this case and I think the significance of a 'content.path' option might easily be more clear to new users than for a 'resource-dir' option.  Is there some notable way that you are thinking 'content.path' would be inaccurate/misleading, or is there some other concern?

File: UI/CUIWnd.cpp [hunk 3]
```diff
@@ -633,32 +634,32 @@ void CUIWnd::SetDefaultedOptions() {
 
 void CUIWnd::SaveDefaultedOptions() {
     OptionsDB& db = GetOptionsDB();
-    std::string config_prefix = "UI.windows." + m_config_name;
-    std::string windowed = "";
-    if (!db.Get<bool>("fullscreen"))
-        windowed = "-windowed";
+    std::string config_prefix = "ui." + m_config_name;
+    std::string windowed = ".fullscreen";
```
Comment (geoffthemedio) [inline UI/CUIWnd.cpp]: ?

File: AI/AIInterface.h (hunk 1 above)
Comment (dbenage-cx) [reply AI/AIInterface.h]: The reason for picking `content` was due to references using the term "game content" as well as aligning with "content scripting" (focs). I'm happy to change the group to `resource.*` if that is the preference.

In either case, I need to update README files (was not searching *.md).

File: UI/CUIWnd.cpp (hunk 3 above)
Comment (dbenage-cx) [reply UI/CUIWnd.cpp]: I don't understand the question ;)
Would the following address the concern: `std::string window_mode = db.Get<bool>("video.fullscreen.enabled") ? ".fullscreen" : ".windowed";`, or is the code unclear and need commenting?

File: AI/AIInterface.h (hunk 1 above)
Comment (dbenage-cx) [reply AI/AIInterface.h]: Pushing this change without squashing, so that it may be easily dropped if decided.

File: UI/CUIWnd.cpp (hunk 3 above)
Comment (geoffthemedio) [reply UI/CUIWnd.cpp]: It seems strange to have a string variable named `windowed` contain the text `fullscreen`. The suggested modification is better.

File: UI/ClientUI.cpp [hunk 4]
```diff
@@ -476,101 +476,100 @@ namespace {
...
     void AddOptions(OptionsDB& db) {
-        db.Add("show-fps",              UserStringNop("OPTIONS_DB_SHOW_FPS"),              false);
-        db.Add("limit-fps",             UserStringNop("OPTIONS_DB_LIMIT_FPS"),             true);
-        db.Add("max-fps",               UserStringNop("OPTIONS_DB_MAX_FPS"),               60.0,
-               RangedStepValidator<double>(1.0, 1.0, 240.0));
-        db.Add("limit-fps-no-focus",    UserStringNop("OPTIONS_DB_LIMIT_FPS_NO_FOCUS"),    true);
-        db.Add("max-fps-no_focus",      UserStringNop("OPTIONS_DB_MAX_FPS_NO_FOCUS"),      15.0,
+        db.Add("video.fps.shown",       UserStringNop("OPTIONS_DB_SHOW_FPS"), false);
+        db.Add("video.fps.max.enabled", UserStringNop("OPTIONS_DB_LIMIT_FPS"),             true);
+        db.Add("video.fps.max",         UserStringNop("OPTIONS_DB_MAX_FPS"),               60.0,
```
Comment (geoffthemedio) [inline UI/ClientUI.cpp]: might be better to call this `video.fps.limit` than `video.fps.max`, but not a major issue...

File: UI/ClientUI.cpp [hunk 5]
```diff
@@ -476,101 +476,100 @@ namespace {
...
+        db.Add("audio.music.volume",                            UserStringNop("OPTIONS_DB_MUSIC_VOLUME"),                      127,                    RangedValidator<int>(1, 255));
+        db.Add("audio.effects.volume",                          UserStringNop("OPTIONS_DB_UI_SOUND_VOLUME"),                   255,                    RangedValidator<int>(0, 255));
+        db.Add<std::string>("ui.control.button.rollover.sound.path", UserStringNop("OPTIONS_DB_UI_SOUND_BUTTON_ROLLOVER"),     (GetRootDataDir() / "default" / "data" / "sound" / "button_rollover.ogg").string());
+        db.Add<std::string>("ui.control.button.click.sound.path", UserStringNop("OPTIONS_DB_UI_SOUND_BUTTON_CLICK"),           (GetRootDataDir() / "default" / "data" / "sound" / "button_click.ogg").string());
+        db.Add<std::string>("ui.control.button.turn.click.sound.path", UserStringNop("OPTIONS_DB_UI_SOUND_TURN_BUTTON_CLICK"), (GetRootDataDir() / "default" / "data" / "sound" / "turn_button_click.ogg").string());
```
Comment (geoffthemedio) [inline UI/ClientUI.cpp]: not sure about this... `turn.button` might be better... it's not a "turn click" for all buttons, but rather then click sound for turn buttons specifically.

Comment (geoffthemedio) [reply UI/ClientUI.cpp]: could get rid of all the `control` as well... seems unnecessary.

File: UI/ClientUI.cpp [hunk 6]
```diff
@@ -476,101 +476,100 @@ namespace {
...
+        db.Add<std::string>("ui.input.keyboard.sound.path",     UserStringNop("OPTIONS_DB_UI_SOUND_TEXT_TYPING"),              (GetRootDataDir() / "default" / "data" / "sound" / "text_typing.ogg").string());
+        db.Add<std::string>("ui.window.minimize.sound.path",    UserStringNop("OPTIONS_DB_UI_SOUND_WINDOW_MAXIMIZE"),          (GetRootDataDir() / "default" / "data" / "sound" / "window_maximize.ogg").string());
+        db.Add<std::string>("ui.window.maximize.sound.path",    UserStringNop("OPTIONS_DB_UI_SOUND_WINDOW_MINIMIZE"),          (GetRootDataDir() / "default" / "data" / "sound" / "window_minimize.ogg").string());
+        db.Add<std::string>("ui.window.close.sound.path",       UserStringNop("OPTIONS_DB_UI_SOUND_WINDOW_CLOSE"),             (GetRootDataDir() / "default" / "data" / "sound" / "window_close.ogg").string());
+        db.Add<std::string>("ui.messagebox.sound.path",  UserStringNop("OPTIONS_DB_UI_SOUND_ALERT"),                    (GetRootDataDir() / "default" / "data" / "sound" / "alert.ogg").string());
```
Comment (geoffthemedio) [inline UI/ClientUI.cpp]: `messagebox` isn't obviously related to alert popups... maybe could be clearer...

File: UI/ClientUI.cpp [hunk 7]
```diff
@@ -476,101 +476,100 @@ namespace {
...
+        db.Add<std::string>("ui.map.fleet.button.rollover.sound.path", UserStringNop("OPTIONS_DB_UI_SOUND_FLEET_BUTTON_ROLLOVER"), (GetRootDataDir() / "default" / "data" / "sound" / "fleet_button_rollover.ogg").string());
```
Comment (geoffthemedio) [inline UI/ClientUI.cpp]: could do without the `map` in these....

File: UI/ClientUI.cpp [hunk 8]
```diff
@@ -476,101 +476,100 @@ namespace {
...
-        db.Add("UI.researchable-tech",          UserStringNop("OPTIONS_DB_UI_RESEARCHABLE_TECH"),          GG::Clr(48, 48, 48, 255),    Validator<GG::Clr>());
-        db.Add("UI.researchable-tech-border",   UserStringNop("OPTIONS_DB_UI_RESEARCHABLE_TECH_BORDER"),   GG::Clr(164, 164, 164, 255), Validator<GG::Clr>());
-        db.Add("UI.unresearchable-tech",        UserStringNop("OPTIONS_DB_UI_UNRESEARCHABLE_TECH"),        GG::Clr(30, 30, 30, 255),    Validator<GG::Clr>());
-        db.Add("UI.unresearchable-tech-border", UserStringNop("OPTIONS_DB_UI_UNRESEARCHABLE_TECH_BORDER"), GG::Clr(86, 86, 86, 255),    Validator<GG::Clr>());
-        db.Add("UI.tech-progress-background",   UserStringNop("OPTIONS_DB_UI_TECH_PROGRESS_BACKGROUND"),   GG::Clr(72, 72, 72, 255),    Validator<GG::Clr>());
-        db.Add("UI.tech-progress",              UserStringNop("OPTIONS_DB_UI_TECH_PROGRESS"),              GG::Clr(40, 40, 40, 255),    Validator<GG::Clr>());
+        db.Add("ui.research.status.completed.background.color", UserStringNop("OPTIONS_DB_UI_KNOWN_TECH"), GG::Clr(72, 72, 72, 255), Validator<GG::Clr>());
```
Comment (geoffthemedio) [inline UI/ClientUI.cpp]: could remove the `status` ...?

File: UI/ClientUI.cpp [hunk 9]
```diff
@@ -476,101 +476,100 @@ namespace {
...
-        db.Add("UI.tooltip.extended-delay",     UserStringNop("OPTIONS_DB_UI_TOOLTIP_LONG_DELAY"),         3500,       RangedValidator<int>(0, 30000));
-        db.Add("UI.multiple-fleet-windows",     UserStringNop("OPTIONS_DB_UI_MULTIPLE_FLEET_WINDOWS"),     false);
-        db.Add("UI.window-quickclose",          UserStringNop("OPTIONS_DB_UI_WINDOW_QUICKCLOSE"),          true);
-        db.Add("UI.auto-reposition-windows",    UserStringNop("OPTIONS_DB_UI_AUTO_REPOSITION_WINDOWS"),    true);
+        db.Add("ui.tooltip.delay",              UserStringNop("OPTIONS_DB_UI_TOOLTIP_DELAY"),              500,        RangedValidator<int>(0, 3000));
+        db.Add("ui.tooltip.extended.delay",     UserStringNop("OPTIONS_DB_UI_TOOLTIP_LONG_DELAY"),         3500,       RangedValidator<int>(0, 30000));
+        db.Add("ui.fleet.multiple.enabled", UserStringNop("OPTIONS_DB_UI_MULTIPLE_FLEET_WINDOWS"),         false);
+        db.Add("ui.quickclose.enabled",  UserStringNop("OPTIONS_DB_UI_WINDOW_QUICKCLOSE"),                 true);
+        db.Add("ui.reposition.enabled",  UserStringNop("OPTIONS_DB_UI_AUTO_REPOSITION_WINDOWS"),           true);
```
Comment (geoffthemedio) [inline UI/ClientUI.cpp]: include the `auto` in the name... otherwise is misleading.

File: UI/FleetWnd.cpp [hunk 10]
```diff
@@ -386,16 +386,18 @@ namespace {
...
-        db.Add("UI.fleet-wnd-scanline-clr", UserStringNop("OPTIONS_DB_UI_FLEET_WND_SCANLINE_CLR"),  GG::Clr(24, 24, 24, 192),   Validator<GG::Clr>());
+        db.Add("ui.fleet.aggression", UserStringNop("OPTIONS_DB_FLEET_WND_AGGRESSION"), INVALID_FLEET_AGGRESSION, Validator<NewFleetAggression>());
+        db.Add("ui.fleet.scanline.color", UserStringNop("OPTIONS_DB_UI_FLEET_WND_SCANLINE_CLR"), GG::Clr(24, 24, 24, 192), Validator<GG::Clr>());
     }
     bool temp_bool = RegisterOptions(&AddOptions);
 
-    NewFleetAggression NewFleetsAggressiveOptionSetting()
-    { return GetOptionsDB().Get<NewFleetAggression>("UI.fleet-wnd-aggression"); }
+    NewFleetAggression NewFleetsAggressiveOptionSetting() {
+        return GetOptionsDB().Get<NewFleetAggression>("ui.fleet.aggression");
```
Comment (geoffthemedio) [inline UI/FleetWnd.cpp]: leave as single line

File: UI/MapWnd.cpp [hunk 11]
```diff
@@ -1816,7 +1824,7 @@ void MapWnd::Render() {
 }
 
 void MapWnd::RenderStarfields() {
-    if (!GetOptionsDB().Get<bool>("UI.galaxy-starfields"))
+    if (!GetOptionsDB().Get<bool>("ui.map.system.background.starfields.shown"))
```
Comment (geoffthemedio) [inline UI/MapWnd.cpp]: this option names doesn't make sense... background starfields have nothing to do with systems

File: UI/ClientUI.cpp (hunk 4 above)
Comment (dbenage-cx) [reply UI/ClientUI.cpp]: I should address other uses of `limit` and `max`/`min` with unified terms. Currently favor `min` over something like `limit.lower`, but will look for alternatives.

File: UI/ClientUI.cpp (hunk 5 above)
Comment (dbenage-cx) [reply UI/ClientUI.cpp]: `turn` is a specific `button`, options of which should take precedence over more generic options.
`click` describes the action, not the sound, I will change to `press`.

Will review if dropping `control` for specific control elements conflicts with specific `window` or generic `ui` elements (specific `window` being previously dropped).

File: UI/ClientUI.cpp (hunk 6 above)
Comment (dbenage-cx) [reply UI/ClientUI.cpp]: `ui.dialog.alert.sound.path`?

File: UI/ClientUI.cpp (hunk 7 above)
Comment (dbenage-cx) [reply UI/ClientUI.cpp]: These are options related to fleets within the map window.  Dropping `map` would conflict or create confusion with `ui.fleet`, options related to fleet window (formerly `ui.window.fleet`).

File: UI/ClientUI.cpp (hunk 8 above)
Comment (dbenage-cx) [reply UI/ClientUI.cpp]: Would prefer to keep the status elements grouped together, helpful when editing/reviewing a config file, though there are other non-realized concerns as well (improved --help, generic status options, procedural options wnd).

File: UI/ClientUI.cpp (hunk 9 above)
Comment (dbenage-cx) [reply UI/ClientUI.cpp]: Prefer `auto_reposition.enabled` or `reposition.auto.enabled`?

File: UI/MapWnd.cpp [hunk 12]
```diff
@@ -6960,89 +6972,89 @@ namespace {
 void MapWnd::ConnectKeyboardAcceleratorSignals() {
     HotkeyManager* hkm = HotkeyManager::GetManager();
 
-    hkm->Connect(boost::bind(&MapWnd::ReturnToMap, this), "map.return_to_map",
+    hkm->Connect(boost::bind(&MapWnd::ReturnToMap, this), "ui.map.open",
                  AndCondition({VisibleWindowCondition(this), NoModalWndsOpenCondition}));
-    hkm->Connect(boost::bind(&MapWnd::OpenMessages, this), "map.open_chat",
+    hkm->Connect(boost::bind(&MapWnd::OpenMessages, this), "ui.map.messages.open",
```
Comment (geoffthemedio) [inline UI/MapWnd.cpp]: this is named inconsistently with the rest...

File: UI/ProductionWnd.cpp [hunk 13]
```diff
@@ -34,9 +34,9 @@ namespace {
 
     void AddOptions(OptionsDB& db) {
         // queue width used also on research screen. prevent double-adding...
-        if (!db.OptionExists("UI.queue-width"))
-            db.Add("UI.queue-width",                    UserStringNop("OPTIONS_DB_UI_QUEUE_WIDTH"),         350,    RangedValidator<int>(200, 500));
-        db.Add("UI.show-production-location-on-queue",  UserStringNop("OPTIONS_DB_UI_PROD_QUEUE_LOCATION"), true,   Validator<bool>());
+        if (!db.OptionExists("ui.queue.width"))
+            db.Add("ui.queue.width", UserStringNop("OPTIONS_DB_UI_QUEUE_WIDTH"), 350, RangedValidator<int>(200, 500));
+        db.Add("ui.queue.source.shown", UserStringNop("OPTIONS_DB_UI_PROD_QUEUE_LOCATION"), true, Validator<bool>());
```
Comment (geoffthemedio) [inline UI/ProductionWnd.cpp]: production location != "source"

File: UI/SaveFileDialog.cpp [hunk 14]
```diff
@@ -72,22 +72,24 @@ namespace {
         // List the columns to show, separated by colons.
         // Valid: time, turn, player, empire, systems, seed, galaxy_age, galaxy_shape, planet_freq, native_freq, specials_freq, starlane_freq
         // These settings are not visible in the options panel; the defaults should be good for regular users.
-        db.Add<std::vector<std::string>>("UI.save-file-dialog.columns", UserStringNop("OPTIONS_DB_UI_SAVE_DIALOG_COLUMNS"),
-                                          StringToList("time,turn,player,empire,file"));
-        db.Add<std::string>("UI.save-file-dialog.time." + WIDE_AS, UserStringNop("OPTIONS_DB_UI_SAVE_DIALOG_COLUMN_WIDE_AS"),
+        db.Add<std::vector<std::string>>("ui.save.columns", UserStringNop("OPTIONS_DB_UI_SAVE_DIALOG_COLUMNS"),
```
Comment (geoffthemedio) [inline UI/SaveFileDialog.cpp]: ambiguous... add `dialog`

File: UI/SystemIcon.cpp [hunk 15]
```diff
@@ -28,8 +28,10 @@
 #include <boost/format.hpp>
 
 namespace {
-    bool PlaySounds()                   { return GetOptionsDB().Get<bool>("UI.sound.enabled"); }
-    void PlaySystemIconRolloverSound()  { if (PlaySounds()) Sound::GetSound().PlaySound(GetOptionsDB().Get<std::string>("UI.sound.system-icon-rollover")); }
+    bool PlaySounds() {
```
Comment (geoffthemedio) [inline UI/SystemIcon.cpp]: one line

File: client/human/HumanClientApp.cpp [hunk 16]
```diff
@@ -174,16 +174,16 @@ namespace {
         }
 
         // only execute default option setting once
-        if (GetOptionsDB().Get<bool>("checked-gl-version"))
+        if (!GetOptionsDB().Get<bool>("version.gl.check.enabled"))
```
Comment (geoffthemedio) [inline client/human/HumanClientApp.cpp]: `enabled` is misleading. `done` would be better

File: util/OptionsDB.cpp [hunk 17]
```diff
@@ -507,10 +507,11 @@ void OptionsDB::SetFromFile(const boost::filesystem::path& file_path,
...
         if (ifs) {
             doc.ReadDoc(ifs);
-            if (version.empty() || (doc.root_node.ContainsChild("version-string") &&
-                                    doc.root_node.Child("version-string").Text() == version))
+            if (version.empty() || (doc.root_node.ContainsChild("version") &&
+                                    doc.root_node.Child("version").ContainsChild("string") &&
+                                    version == doc.root_node.Child("version").Child("string").Text()))
             {
-                GetOptionsDB().SetFromXML(doc);
+                    GetOptionsDB().SetFromXML(doc);
```
Comment (geoffthemedio) [inline util/OptionsDB.cpp]: unnecessary extra indent

File: UI/ClientUI.cpp (hunk 6 above)
Comment (geoffthemedio) [reply UI/ClientUI.cpp]: lose the `dialog` ?

File: UI/ClientUI.cpp (hunk 4 above)
Comment (geoffthemedio) [reply UI/ClientUI.cpp]: limits and maxes aren't necessarily synonymous ...

File: UI/MapWnd.cpp (hunk 12 above)
Comment (dbenage-cx) [reply UI/MapWnd.cpp]: `ui.map.messages.open.hotkey` is distinct from `ui.map.messages.hotkey`. The former forces the window open, the later toggles the window open or closed.

If no objection, I will remove the former.

File: UI/ClientUI.cpp (hunk 9 above)
Comment (geoffthemedio) [reply UI/ClientUI.cpp]: consistent use of . seems better

File: UI/MapWnd.cpp (hunk 12 above)
Comment (geoffthemedio) [reply UI/MapWnd.cpp]: Several other window toggle command exist and have hotkeys... particularly those with top-right buttons... so not having one for messages may be odd. (if that's what you're suggesting...)

**Comment by geoffthemedio** [comment]
otherwise OK... Or at least enough so given the amount of additional time I can(n't) spend looking over it all again.

File: UI/MapWnd.cpp (hunk 12 above)
Comment (dbenage-cx) [reply UI/MapWnd.cpp]: To clarify: A hotkey option to toggle messages window open/closed will still exist as `ui.map.messages.hotkey`

In current master there are two hotkeys for this window:
*  [`UI.hotkeys.map.messages `](https://github.com/freeorion/freeorion/blob/master/UI/MapWnd.cpp#L173) : Toggles window open/closed
*  [`UI.hotkeys.map.open_chat `](https://github.com/freeorion/freeorion/blob/master/UI/MapWnd.cpp#L166) : Opens window

I've removed the later (and `MapWnd::OpenMessages`, as it is now unused)

**Comment by dbenage-cx** [comment]
@geoffthemedio  TY for all of the reviewing!

Barring any new change requests, may be a couple of days before merge (to retest options after squash/update/rebase).

File: UI/MapWnd.cpp (hunk 12 above)
Comment (geoffthemedio) [reply UI/MapWnd.cpp]: ok

**Pull request merged**